### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by cjorda15**
--What's this PR do?

-mainly moved api calls stemming from the article component to a separate imported file to make it more readable as well as threw away code that didn't seem necessary and trimmed others. 
-also made the router a bit more dynamic in that a user will never be able to go on forward, backward, or on refresh seeing routes of the page they shouldn't while being logged in or not. 

--Where should the reviewer start?

-top to bottom, is it easier to scan the refactored article with the given api helper function names rather than having the api calls living in the article or should I rename some that aren't so clear what they are doing.
-Also curious if the routes make any sense/are readable from a quick scan with the logic and insertion of the redirect functionality on the content component

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "presets": ["es2015", "react"],
+  "presets": [ "es2015", "react"],
```
Comment (letakeane) [inline .babelrc:2]: Not sure if this is a linting thing; I'd either add a space before the closing bracket to match, or take out the space after the opening bracket

File: app/Content/Content.js [hunk 2]
```diff
@@ -34,10 +34,26 @@ render(){
...
+             :
+           <LogInContainer history={history}/> }}/>
+          <Route path="/signup" render={({ history }) => {
+            return this.props.user?
+            <Redirect to='/' />
+              :
+            <SignUpContainer history={history}/>}}/>
+          <Route path="/favorites" render={({history}) => {
+            return !this.props.user?
+            <Redirect to='/' />
```
Comment (letakeane) [inline app/Content/Content.js:54]: THIS IS SO COOL

File: app/Content/components/Article/apiHelpers.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
+export const apiShowFavorites = (user,handleFunction) => {
```
Comment (letakeane) [inline app/Content/components/Article/apiHelpers.js:1]: Yeahhhh pulling those functions out into a separate file!

File: src/app.js [hunk 4]
```diff
@@ -120,13 +120,13 @@ app.put('/api/v1/news',(req,res) => {
...
 })
 
-app.post('/api/v1/user', (request, response) => {
+app.post('/api/v1/user', (request, response, next) => {
   const user = request.body;
   database('user').insert(user,'id')
-   .then(user => {response.send(user)})
-   .then(response => console.log(response))
+   .then(user => { console.log('WOOT', user); response.send(user)})
+  //  .then(response => console.log(response))
```
Comment (letakeane) [inline src/app.js:127]: Don't forget to kill these comments when you don't need them anymore

File: .babelrc [hunk 5]
```diff
@@ -1,5 +1,5 @@
...
   /*"plugins": ["css-modules-transform"]*/
```
Comment (Adamj1232) [inline .babelrc]: Saving these for later?

File: app/Content/Content.js [hunk 6]
```diff
@@ -3,7 +3,7 @@ import {buildList, showFavorties, addArticles} from './actions'
 import React, {Component} from 'react';
```
Comment (Adamj1232) [inline app/Content/Content.js:3]: Might be good to add some spacing within the brackets to be consistent and match line 6

File: app/Content/components/Article/Article.js [hunk 7]
```diff
@@ -45,127 +35,43 @@ class Article extends Component {
...
-        updated_at:month+" "+day+" "+year
-      })
-    })
-    .then(response => response.json())
-    .then(response => this.handleResponse(response))
-    .catch(error => console.log(error,"error message")))
-     :
-    (this.setState({bottomCardMessage:"Please log in to save a article",showInfo:false}),
-    this.handleError())
+    apiAddFavorites(this.props.user,this.props.article,this.props.source,this.handleResponse.bind(this),this.setState.bind(this),this.handleError.bind(this))
```
Comment (Adamj1232) [inline app/Content/components/Article/Article.js:52]: Damn thats a sweet reFACTOR!

**Comment by Adamj1232** [comment]
Some sweet refactoring and cleanup!! 
It would be good to choose weather you want spaces within your curly bracket and then be consistent through out

File: src/app.js (hunk 4 above)
Comment (cjorda15) [reply src/app.js:127]: ahh everytime.. good catch!

File: app/Content/Content.js (hunk 6 above)
Comment (cjorda15) [reply app/Content/Content.js:3]: ah this is true, danka!

File: .babelrc (hunk 1 above)
Comment (cjorda15) [reply .babelrc:2]: good point!

**Pull request merged**